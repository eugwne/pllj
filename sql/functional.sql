do $$
local f = require('pllj.func')
local fn = f.find_function('quote_nullable(text)')
print(fn("qwerty"))
print(fn(nil))
$$ language pllj;

create or replace function pg_temp.get_json() returns jsonb as $$
select '{"a":5, "b":10}'::jsonb
$$ language sql;

do $$
    local fn = require('pllj.func').find_function('pg_temp.get_json()')
    _G.get_json = fn
    print(get_json())
$$ language pllj;

create or replace function pg_temp.get_json() returns jsonb as $$
select '{"a":50, "b":100}'::jsonb
$$ language sql;

do $$
print(get_json())
$$ language pllj;

drop function pg_temp.get_json();

do $$
    local _, e = pcall(get_json)
    print(string.find(e, "cache lookup failed for function")~=nil)
$$ language pllj;