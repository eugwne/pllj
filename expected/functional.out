do $$
local f = require('pllj.func')
local fn = f.find_function('quote_nullable(text)')
print(fn("qwerty"))
print(fn(nil))
$$ language pllj;
INFO:  'qwerty'
INFO:  NULL
create or replace function pg_temp.get_json() returns jsonb as $$
select '{"a":5, "b":10}'::jsonb
$$ language sql;
do $$
    local fn = require('pllj.func').find_function('pg_temp.get_json()')
    _G.get_json = fn
    print(get_json())
$$ language pllj;
INFO:  {"a": 5, "b": 10}
create or replace function pg_temp.get_json() returns jsonb as $$
select '{"a":50, "b":100}'::jsonb
$$ language sql;
do $$
print(get_json())
$$ language pllj;
INFO:  {"a": 50, "b": 100}
drop function pg_temp.get_json();
do $$
    local _, e = pcall(get_json)
    print(string.find(e, "cache lookup failed for function")~=nil)
$$ language pllj;
INFO:  true
